src/recipe-selector/recipe-selector.ts

```diff
@@ -62,3 +63,29 @@ export class RecipeSelector {
       return null;
     }
 }
+
+export class SimplePlanner {
+  private _selector: RecipeSelector;
+  
+  constructor(readonly recipes: Recipe[]) {
+    this._selector = new RecipeSelector(recipes);
+  }
+  
+  async plan(arc: Arc, request:  [string, string][]): Promise<Recipe> {
+    const resolver = new RecipeResolver(arc);
+    const recipe = this._selector.select(request);
+    if (recipe) {
+      const options: IsValidOptions = {errors: new Map()};
+      const resolved = await resolver.resolve(recipe, options);
+      if (resolved) {
+        return resolved;
+      } else {
+        console.log(`could not resolve selected recipe: ${[...options.errors.values()].join('\n')}`);
+        throw new Error('Resolution failed');
```

Comment:
i don't think either of the negative cases - not found or unresolved recipes should be a throw, i think returning null is better. it is a valid use case for a recipe not being found or resolved.